### ISSUE TITLE
Is this gem maintained?

### ISSUE THREAD

**Issue by Preen** [capistrano-plugins/capistrano-unicorn-nginx#103]
Seems pretty dead.

**Comment by rhomeister** [comment]
It isn't actively maintained no. I'm one of the authors and this gem currently satisfies my purposes. However, I'd strongly welcome more contributors. So if you're interested, please let me know.

**Comment by griffithac** [comment]
@rhomeister thanks for the invite, I use the gem everyday and would be happy to help in any way I can.  I will look at the open issues and see what I can do.  If you have any requests I can always give it a shot.

**Comment by griffithac** [comment]
@rhomeister, I am currently working on the `systemd` branch as I and other users need support for Ubuntu 16.04 and other distros.  Once I/we are finished and have tested that there are no breaking changes (to 14.04 for example), what would I need to do to get a version 4.1 of the gem released?

**Comment by Preen** [comment]
@rhomeister can you invite @mileandra as collaborator too as she is part of our team and wants to help out :).

**Comment by dennislysenko** [comment]
@griffithac so, I'm on ubuntu 16.04 and notice that about half the time unicorn fails to start but I can't track down any errors in the logs when this happens. Is this what you are referring to with 16.04 support? Also, I still have many servers on 14.04 which I could use to help field-test that it will not break existing 14.04 setups.

**Comment by griffithac** [comment]
@dennislysenko, yes I was having the same problem.  I am using my own branch in production with 16.04 right now and it is working.  I want to merge my changes but like you mentioned I was worried about backwards compatibility.

The problem is with the change to systemd from upstart with Ubuntu 16.04.

I currently using my `systemd` branch at https://github.com/griffithac/capistrano-unicorn-nginx.git

**Comment by griffithac** [comment]
Also I need to test that the setup process works for a totally new server install.  I am not 100% confident that other changes I made to my server were not part of the fix.

I need to find some more time to put into it.

**Comment by rhomeister** [comment]
Hi all, guaranteeing that the gem works with Ubuntu 14.04 and 16.04 remains an essential goal. Our servers are running on 14.04, and I've read that many users of this gem are using that version as well. Other than that, I completely understand the need for supporting 16.04. Since this is not a priority for me personally though, you shouldn't count on my effort to support this version. However, I'd love to have more people working on this project to achieve that goal. To this end, I've just added @mileandra as an collaborator. Welcome on board!

**Comment by griffithac** [comment]
@rhomeister thanks.  @mileandra, welcome.

I just opened a PR.  It has the code that is working for me on 16.04.  We don't need to use my code, but I thought that it may be helpful to others working on the project.

**Comment by Preen** [comment]
What we ould do besides the systemd is adding instructions to the readme on how to replace systemd with upstart. Not that its a long term solution.

```Install the upstart-sysv package, which will remove ubuntu-standard and systemd-sysv (but should not remove anything else -- if it does, yell!), and run sudo update-initramfs -u.```

**Comment by griffithac** [comment]
That is not a bad idea at all. Not optimal, but it may get some users down the road. Have you tested that doing this with Ubuntu 16.04 and the current gem works?

Andrew Griffith
Griffith Industries, Inc.
206-786-7440
andrewg@griffithind.com

**Comment by Preen** [comment]
Yes! As a matter of fact I did run into this problem just this morning and it worked out good! The start process is a bit slow but it works. Restart is still fast.

**Comment by dennislysenko** [comment]
@griffithac thanks for the pointer, I switched to your branch and it works perfectly. I doubt it's just fixes you did on your server--I reproduced it across two separate client deployments, one of which was created just 3 days ago and one 1-2 weeks ago. Both are on 16.04. I will drop a line here if I get a chance to test it with another deployment composed of 14.04 servers.

**Comment by dennislysenko** [comment]
Hey @griffithac, heads up, on line 20 of the unicorn_init.erb generator, you'll want to move the .to_command outside of the last parenthesis--makes it impossible to run unicorn:setup_initializer without hotpatching that as it's trying to call to_command on the last arg to the command rather the command itself :)

**Comment by tomprats** [comment]
I haven't seen much activity in this gem since the new maintainers were added. Are you guys still onboard?

I just started using Capistrano at the end of last year, but will be using it and this gem if it can/will be maintained. I'll drop and comment on PRs when I can test them. Let me know if you'd be willing to add me as a collaborator.

I've created two PRs in reference to this issue I found: https://github.com/capistrano-plugins/capistrano-unicorn-nginx/issues/107

I also will be using other [capistrano-plugins](https://github.com/capistrano-plugins) organization gems and am actively willing to contribute to them as well (https://github.com/capistrano-plugins/capistrano-postgresql/pull/48)

Thanks!

**Comment by Preen** [comment]
@tomprats Thats awesome. I will try to keep up with your speed :) As for you being an maintainer. Im all for it!
